src/Kestrel.Core/Internal/Infrastructure/HttpCharacters.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure
+{
+    internal static class HttpCharacters
+    {
+        public static readonly int TableSize = 128;
+        public static readonly bool[] AlphaNumeric;
+        public static readonly bool[] Authority;
```

Comment:
private

Reply:
Some of these arrays may be used as is instead of via the IndexOf methods: https://github.com/aspnet/KestrelHttpServer/pull/2707/files#diff-d3b42025fb7322d45d6c332b7723177eR447